ethcore/src/service.rs

```diff
@@ -146,6 +148,32 @@ impl ClientService {
 	pub fn add_notify(&self, notify: Arc<ChainNotify>) {
 		self.client.add_notify(notify);
 	}
+
+	#[cfg(feature="stratum")]
+	/// Runs stratum if feature is on
+	pub fn stratum_notifier(cfg: &StratumOptions, miner: Weak<Miner>, client: Weak<Client>) -> Result<Box<miner::NotifyWork>, ()> {
+		use miner::Stratum;
+		use miner::NotifyWork;
+
+		Stratum::new(&cfg.io_path, miner, client).or_else(|e|
+		{
+			warn!(target: "stratum", "Cannot start stratum server: {:?}", e);
+			Err(e)
+		})
+		.map(|stratum|
+		{
+			stratum.run_async();
+			Box::new(stratum) as Box<NotifyWork>
+		})
+		// Warnings produced upstream, this is just conditional compilation workaround
+		.map_err(|_|())
```

Comment:
Why not display warnings here? Or even propagate errors upstream (even as `String`)